src/new/wallet.js

```diff
@@ -2666,6 +2675,168 @@ class HathorWallet extends EventEmitter {
+    }
+
+    // Get the blueprint data from full node
+    const blueprintInformation = await ncApi.getBlueprintInformation(blueprintId);
+    const methodArgs = get(blueprintInformation, `public_methods.${method}.args`);
+    if (!methodArgs) {
+      throw new NanoContractTransactionError(`Blueprint does not have method ${method}.`);
+    }
+
+    // Args may come as undefined
+    const argsLen = data.args ? data.args.length : 0;
+    if (argsLen !== methodArgs.length) {
+      throw new NanoContractTransactionError(`Method needs ${methodArgs.length} parameters but data has ${data.args.length}.`);
+    }
+
+    // Here we validate that the arguments sent in the data array of args has
+    // the expected type for each parameter of the blueprint method
+    for (const [index, arg] of methodArgs.entries()) {
+      let typeToCheck = arg.type;
+      if (typeToCheck.startsWith('SignedData')) {
+        // Signed data will always be an hexadecimal with the
+        // signed part and the data itself
+        typeToCheck = 'str';
+      }
+      switch (typeToCheck) {
+        case 'bytes':
+          // Bytes arguments are sent in hexadecimal
+          try {
+            data.args[index] = hexToBuffer(data.args[index]);
+          } catch {
+            // Data sent is not a hex
+            throw new NanoContractTransactionError(`Invalid hexadecimal for argument number ${index + 1}.`);
+          }
+          break;
+        case 'int':
+        case 'float':
+          if (typeof data.args[index] !== 'number') {
+            throw new NanoContractTransactionError(`Expects argument number ${index + 1} type ${arg.type} but received type ${typeof data.args[index]}.`);
+          }
+          break;
+        case 'str':
+          if (typeof data.args[index] !== 'string') {
+            throw new NanoContractTransactionError(`Expects argument number ${index + 1} type ${arg.type} but received type ${typeof data.args[index]}.`);
+          }
+          break;
+        default:
+          if (arg.type !== typeof data.args[index]) {
+            throw new NanoContractTransactionError(`Expects argument number ${index + 1} type ${arg.type} but received type ${typeof data.args[index]}.`);
+          }
+      }
+    }
+
+    // Get private key that will sign the nano contract transaction
+    let derivedKey;
+    try {
+      derivedKey = await this.getHDPrivateKeyFromAddress(address, options);
+    } catch (e) {
+      if (e instanceof AddressError) {
+        throw new NanoContractTransactionError('Address used to sign the transaction does not belong to the wallet.');
+      }
+      throw e;
+    }
+    const privateKey = derivedKey.privateKey;
+
+    // Build and send transaction
+    const builder = new NanoContractTransactionBuilder();
+    builder.setMethod(method);
+    builder.setWallet(this);
+    builder.setBlueprintId(blueprintId);
+    builder.setNcId(method === 'initialize' ? null : data.ncId);
```

Comment:
Isn't data.ncId already null if the method is initialize?
If not, maybe we can create a local ncId and just use it instead of having logic here.

Reply:
I did a refactor following your suggestions. Can you check again, please? https://github.com/HathorNetwork/hathor-wallet-lib/pull/604/commits/af4d8a0e5b472a12201b29cdca14d828af8adbf6